### PR TITLE
Added Spring Boot SnapStart Example

### PR DESCRIPTION
**Pull request by muraria**
Added SnapStart example for Spring Boot version of the Example.
Removed unused file in Dagger Load test folder

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,10 +26,11 @@ and the storage logic.
 
 ## Infrastructure
 
-The sample application can be deployed in two different ways:
+The sample application can be deployed in three different ways:
 1. JVM - This mode uses zip packaging style and runs on the JVM inside the Lambda environment.
 2. Custom Runtime - This mode uses GraalVM native image and uses Custom Runtime to run on Lambda. 
    GraalVM native image is a stand-alone execution binary and does not require a JVM to run.
+3. SnapStart - This mode enables SnapStart on the Lambda functions. Current examples only included Spring boot version of the application. Stay tuned for other framework examples with SnapStart. Learn more about SnapStart [here](https://docs.aws.amazon.com/lambda/latest/dg/snapstart.html).
```
Comment (dmahapatro) [inline README.md:33]: JVM (without SnapStart)
JVM (with SnapStart)
Custom Runtime

in that order seems like a good progression

File: springboot/README.md [hunk 2]
```diff
@@ -74,14 +86,20 @@ cd load-test
...
+```
 This is a demanding load test, to change the rate alter the `arrivalRate` value in `load-test.yml`.
 
 ## CloudWatch Logs Insights
 
 Using this CloudWatch Logs Insights query you can analyse the latency of the requests made to the Lambda functions.
 
-The query separates cold starts from other requests and then gives you p50, p90 and p99 percentiles.
+The query separates cold starts from other requests and then gives you p50, p90 and p99 percentiles. 
+Please note that this query is not applicable to SnapStart version.
```
Comment (dmahapatro) [inline springboot/README.md:102]: > ⚠️ Highlight this as a warning using markdown syntax like this

File: springboot/README.md [hunk 3]
```diff
@@ -100,9 +118,44 @@ Latency for GraalVM version:
...
+```
+
+<p align="center">
+  <img src="../imgs/springboot/springboot-snapstart-warm-log-insights.JPG" alt="Cold Start metrics with SnapStart"/>
+</p>
+
 ## AWS X-Ray Tracing
 You can add additional detail to your X-Ray tracing by adding a TracingInterceptor to your AWS SDK clients.
 
+Please note that AWS Lambda SnapStart currently does not support X-ray tracing. For this reason, tracing is disabled for all lambda functions in SnapStart version.
```
Comment (dmahapatro) [inline springboot/README.md:157]: Also mention which AWS regions support SnapStart today. Point to the user guide page, if available

File: springboot/src/main/java/software/amazonaws/example/product/product/PrimingResource.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  public PrimingResource(ProductDao productDao) {
+    this.productDao = productDao;
+    Core.getGlobalContext().register(this);
+  }
+
+  @Override
+  public void beforeCheckpoint(Context<? extends Resource> context) throws Exception {
+    System.out.println("beforeCheckpoint hook");
+    productDao.getAllProduct();
```
Comment (dmahapatro) [inline springboot/src/main/java/software/amazonaws/example/product/product/PrimingResource.java:27]: Why `getAllProduct()` during priming? If there is a valid reason, can this be explained in readme.

**Review by dmahapatro — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (muraria) [reply README.md]: Done

File: springboot/README.md (hunk 2 above)
Comment (muraria) [reply springboot/README.md]: Done

File: springboot/README.md (hunk 3 above)
Comment (muraria) [reply springboot/README.md:158]: Added regions

File: springboot/src/main/java/software/amazonaws/example/product/product/PrimingResource.java (hunk 4 above)
Comment (muraria) [reply springboot/src/main/java/software/amazonaws/example/product/product/PrimingResource.java]: Changed the function to getProduct() and added comments.

File: springboot/.aws-sam/build.toml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# This file is auto generated by SAM CLI build command
```
Comment (dmahapatro) [inline springboot/.aws-sam/build.toml:1]: .aws-sam should be gitignored

File: springboot/src/main/java/software/amazonaws/example/product/product/PrimingResource.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  public PrimingResource(ProductDao productDao) {
+    this.productDao = productDao;
+    Core.getGlobalContext().register(this);
+  }
+
+  @Override
+  public void beforeCheckpoint(Context<? extends Resource> context) throws Exception {
+    System.out.println("beforeCheckpoint hook");
+    //This call to getProduct will initialize the AWS DynamoDb SDK before the snapshot is taken. This technique is called Priming.
+    productDao.getProduct("1");
```
Comment (dmahapatro) [inline springboot/src/main/java/software/amazonaws/example/product/product/PrimingResource.java:28]: Wouldn't describe-table be more efficient here than a getbypartitionkey?

**Review by dmahapatro — CHANGES_REQUESTED** [review]

**Review by dmahapatro — APPROVED** [review]

**Pull request merged**